haas/client/auth.py

```diff
@@ -0,0 +1,32 @@
+import requests
```

Comment:
I tried refactoring it the way you suggested but I think this requires a little more discussion. 
Global variables `http_client` and `C` will have to coexist until all calls have migrated to the client library. 

Lets do it when we meet this week. Will mail you with suitable day and time.